bindings/python/pyiree/compiler/BUILD

```diff
@@ -88,6 +88,7 @@ pybind_cc_library(
     tags = ["nokokoro"],
     deps = COMPILER_DEPS + [
         "//bindings/python/pyiree/common",
+        "//iree/tools:init_compiler_modules",
```

Comment:
Can you please the dep `iree::tools::init_compiler_modules` to the `CMakeLists.txt` as well?

I think we further need to revise the variable `COMPILER_DEPS` in the `BUILD` and the `CMakeLists.txt` file. This already introduces a dep to `TensorListDialect`. This is not mandatory, as we now depend on `init_compiler_modules`, which however is required, as this target includes the headers.

Reply:
Good catch. I thought pyiree is bazel only given it integrates with TensorFlow. But apparently I was wrong.

Also removed `TensorListDialect` from `COMPILER_DEPS`.